### PR TITLE
Fix for slowness due to folds

### PR DESCRIPTION
**Pull request by PalmaITEM**
Larger files suffered from quite some lag and the reason was the folding.

More specifically it seems that the used pattern magic was a big cause of this lag.

### CODE REVIEW

File: ftplugin/markdown.vim [hunk 1]
```diff
@@ -23,9 +23,8 @@ function! MarkdownFold()
   let line = getline(v:lnum)
 
   " Regular headers
-  let depth = match(line, '\(^#\+\)\@<=\( .*$\)\@=')
-  if depth > 0
-    return ">" . depth
+  if (match(line, '^#\+ .*$') == 0)
```
Comment (tpope) [inline ftplugin/markdown.vim]: Please simplify to the more conventional `line =~# '^#\+ '`.

Comment (PalmaITEM) [reply ftplugin/markdown.vim]: Good point, thanks. It should be better now.

File: ftplugin/markdown.vim [hunk 2]
```diff
@@ -23,7 +23,7 @@ function! MarkdownFold()
   let line = getline(v:lnum)
 
   " Regular headers
-  if (match(line, '^#\+ .*$') == 0)
+  if (line =~ '^#\+ .*$')
```
Comment (tpope) [inline ftplugin/markdown.vim]: `=~#`, so the linter doesn't get mad (and you can drop the outer parentheses).

Comment (tpope) [reply ftplugin/markdown.vim]: And `.*$` is redundant (but the old code had the same redundancy, so that one's on me).

Comment (PalmaITEM) [reply ftplugin/markdown.vim]: Sorry for missing the `#`, fixed.

I've also removed the redundant `.*$`.

**Pull request merged**